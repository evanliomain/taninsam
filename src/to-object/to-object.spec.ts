import { describe, expect, test } from 'vitest';
import { toObject } from './to-object';

describe('toObject function', () => {
  const input: [string, number | string][] = [
    ['a', 1],
    ['b', 'c'],
  ];

  test('is a pure function', () => {
    toObject(
      x => `${x}`,
      x => x,
    )(input);
    expect(input).toMatchSnapshot();
  });

  test("[['a', 1], ['b', 'c']] |> toObject(item => item[0], item => item[1]) === {a: 1, b: 'c'}", () => {
    expect(
      toObject<[string, number | string], string | number>(
        item => item[0],
        item => item[1],
      )(input),
    ).toMatchSnapshot();
  });
});
