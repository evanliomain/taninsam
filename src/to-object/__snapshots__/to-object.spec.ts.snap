// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`toObject function > [['a', 1], ['b', 'c']] |> toObject(item => item[0], item => item[1]) === {a: 1, b: 'c'} 1`] = `
{
  "a": 1,
  "b": "c",
}
`;

exports[`toObject function > is a pure function 1`] = `
[
  [
    "a",
    1,
  ],
  [
    "b",
    "c",
  ],
]
`;
